gpflow/posterior.py

```diff
@@ -346,8 +346,14 @@ def _get_Kff(self, Xnew, full_cov):
     def _conditional_with_precompute(
         self, Xnew, full_cov: bool = False, full_output_cov: bool = False
     ) -> MeanAndVariance:
-        # Qinv: [L, M, M]
+        # Qinv: [M, M] or [L, M, M]
         # alpha: [M, L]
+        if self.Qinv.shape.ndims == 2:
+            # When using a shared kernel and shared inducing points, the shape of Qinv is [M, M].
+            # We need to manually reintroduce the axis representing the number of outputs.
+            Qinv = tf.repeat(self.Qinv[None, :], self.kernel.output_dim, axis=0)
```

Comment:
why do we have to repeat this? does broadcasting not work in this context ?

Reply:
Are you suggesting using `broadcast_to` instead of `repeat` here?